department-of-veterans-affairs/caseflow
# Justin/appeals 15631

Resolves [APPEALS-15631](https://vajira.max.gov/browse/APPEALS-15631)

### Description
Added a cancel warning message so that if changes have been made on the Cavc Dashboard, the message will alert the user. If the user clicks the 'x' or the 'cancel' button, the modal will disappear and show the dashboard again. If the user clicks remove, the changes will be removed and the user will return to the case details page. If no changes have been made, the user will return to case details without the warning displaying.

### Acceptance Criteria
- [ ] When clicking Cancel on the CAVC Dashboard page users are notified if changes were made
- [ ] When user attempts to leave the page without saving a banner appears:
- [ ] Banner Header: Your changes are not saved
- [ ] Banner text: Please save the changes
- If Ok is clicked in the banner:
- [ ] User is returned to the appeal’s Case Details screen
- [ ] Inputs are not retained – reverts back to it’s last saved state
- [ ] If Cancel is clicked banner is dismissed

### Testing Plan
1. Log in as OAI user and go to a CAVC dashboard on a CAVC Appeal.
2. When initially opening the page, the cancel button should be clickable at the bottom of the page, and should return the user to the case details page.
3. Returning to the dashboard, make a change on the dashboard (add an issue, change disposition, etc)
4. Click the cancel button, and the warning modal should pop up informing user to save changes. 
5. Verify that clicking x or cancel will return user to see the changes they made. 
6. Clicking on "Remove" should cause the changes to be removed and user to return to the case details page.

### User Facing Changes

 BEFORE|AFTER
 No modal (warning message) before
---
|
---
![image](https://user-images.githubusercontent.com/95875751/223584863-fd6ae333-6d70-4088-ab7d-f3cc57fc95fc.png)
